### PR TITLE
New package: cereal-1.3.0

### PR DESCRIPTION
**Pull request by zlice**
https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/cereal

https://github.com/USCiLab/cereal/issues/262

### CODE REVIEW

File: srcpkgs/cereal/template [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+pkgname=cereal
+version=1.3.0
+revision=1
+archs=noarch
+wrksrc=cereal-${version}
+build_style=cmake
+configure_args="-DSKIP_PORTABILITY_TEST=1 -DTHREAD_SAFE=ON -DWITH_WERROR=OFF"
+short_desc="Header-only C++ serialization library"
+maintainer="Orphaned <orphan@voidlinux.org>"
+license="BSD"
```
Comment (Johnnynator) [inline srcpkgs/cereal/template]:
```suggestion
license="BSD-3-Clause"
```

File: srcpkgs/cereal/template [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'cereal'
+pkgname=cereal
+version=1.3.0
+revision=1
+archs=noarch
+wrksrc=cereal-${version}
```
Comment (Johnnynator) [inline srcpkgs/cereal/template]: This is the default value, you can just remove this line.

**Comment by zlice** [comment]
only seeing it fail on license and armv6l-musl BOOTSTRAP=x86_64-musl

`srcpkgs/cereal/template:11: use SPDX id for 'BSD' license or see Manual.md` - not sure what Manual.md it means or what the SPDX id means

i'll wait so you can look at the travis output

**Comment by zlice** [comment]
see the atomic_load_8 musl errors in other projects too. not sure if that will prevent this from being accepted

File: srcpkgs/cereal/template [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+build_style=cmake
+configure_args="-DSKIP_PORTABILITY_TEST=1 -DTHREAD_SAFE=ON -DWITH_WERROR=OFF"
+short_desc="Header-only C++ serialization library"
+maintainer="Orphaned <orphan@voidlinux.org>"
```
Comment (Johnnynator) [inline srcpkgs/cereal/template]: New packages should not be Orphaned

Comment (zlice) [reply srcpkgs/cereal/template]: suggestion ? don't want to put my e-mail in there

**Comment by zlice** [comment]
https://github.com/USCiLab/cereal/issues/591

trying CFLAGS, not sure if that's an acceptable workaround

**Comment by zlice** [comment]
everything failed w/ cflags for some reason (builds on my box fine)
armv6 failed again
let me know on e-mail

File: srcpkgs/cereal/template (hunk 3 above)
Comment (Johnnynator) [reply srcpkgs/cereal/template]: Why not? You already have your email in this commit.

**Comment by Hoshpak** [comment]
Errors like:
```
/usr/lib/gcc/arm-linux-musleabihf/9.2.0/../../../../arm-linux-musleabihf/bin/ld: /usr/arm-linux-musleabihf/usr/include/c++/9.2.0/atomic:254: undefined reference to `__atomic_load_8'
```

Usually mean that you need libatomic for the architecture. This can be done by conditionally adding `libatomic-devel` to makedepends and linking with `-latomic`. You can find an example of this in the libfilezilla template.

**Comment by zlice** [comment]
tried that and the suggestion on the issue i posted. no go

**Comment by Hoshpak** [comment]
It works when you put a space before `-DCMAKE_CXX_STANDARD_LIBRARIES=-latomic`.

**Comment by zlice** [comment]
thanks hoshpak

**Comment by Johnnynator** [comment]
I took the liberty to just change the commit message and manually merge it, since you missed to do so yourself :)

**Comment by zlice** [comment]
sorry, thanks

**Pull request closed without merging**